python/lib/modeldata/dmod/modeldata/subset/hydrofabric_subset.py

```diff
@@ -0,0 +1,217 @@
+        """
+        Get the associated catchments as ::class:`Catchment` objects.
+
+        Returns
+        -------
+        Tuple[Catchment]
+            The associated catchments as ::class:`Catchment` objects.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def nexuses(self) -> Tuple[Nexus]:
+        """
+        Get the associated nexuses as ::class:`Nexus` objects.
+
+        Returns
+        -------
+        Tuple[Catchment]
+            The associated nexuses as ::class:`Nexus` objects.
+        """
+        pass
+
+    @abstractmethod
+    def validate_hydrofabric(self, hydrofabric) -> bool:
+        """
+        Determine whether the given hydrofabric is valid for this subset object.
+
+        A hydrofabric is valid if it has sufficient data to retrieve or derive <i>appropriate</i> ::class:`Catchment`
+        and ::class:`Nexus` objects to represent all the catchments and nexuses of this subset object.
+
+        Note that ::class:`Catchment` and ::class:`Nexus` are listed for simplicity and generality, but implementations
+        may require more specific subtypes of either when determining hydrofabric validity.
+
+        Returns
+        -------
+        bool
+            ``True`` if the hydrofabric's has sufficient data, in format and content, for this subset, or ``False``
+            otherwise.
+        """
+        pass
+
+
+class SimpleHydrofabricSubset(HydrofabricSubset):
+    """
+    Simple ::class:`HydrofabricSubset` type hydrofabric being one or more catchment or nexus objects.
+    """
+
+    @classmethod
+    def factory_create_from_base_and_hydrofabric(cls, subset_def: SubsetDefinition,
+                                                 hydrofabric: Union[Sequence[Union[Catchment, Nexus]], Catchment, Nexus]) \
+            -> 'SimpleHydrofabricSubset':
+        """
+        Convenience method for creating from a simpler subset def object and a hydrofabric.
+
+        Parameters
+        ----------
+        subset_def : SubsetHandler
+            Simple subset definition object, encapsulating the required catchment and nexus ids.
+
+        hydrofabric : Union[Sequence[Catchment, Nexus], Catchment, Nexus]
+            Hydrofabric parameter of an acceptable type.
+
+        Returns
+        -------
+        SimpleHydrofabricSubset
+            A ::class:`SimpleHFSubsetDefinition` object for the same subset defined by ``subset_def``.
+        """
+        return cls(catchment_ids=subset_def.catchment_ids, nexus_ids=subset_def.nexus_ids, hydrofabric=hydrofabric)
+
+    __slots__ = ["_catchments", "_nexuses"]
+
+    def __init__(self, catchment_ids: Collection[str], nexus_ids: Collection[str],
+                 hydrofabric: Union[Sequence[Union[Catchment, Nexus]], Catchment, Nexus]):
+        self._catchments: Set[Catchment] = set()
+        self._nexuses: Set[Nexus] = set()
+        super(SimpleHydrofabricSubset, self).__init__(catchment_ids, nexus_ids, hydrofabric)
```

Comment:
I used that syntax because of my IDE's autocomplete, but I thought that was using `super` to get the right superclass type, then calling its `__init__`.  Is that not right?